src/util/text-tracks.js

```diff
@@ -223,6 +223,60 @@ export const addMetadata = ({
   });
 };
 
+/**
+ * Add daterange metadata text track to a source handler given an array of metadata
+ *
+ * @param {Object}
+ *   @param {Object} inbandTextTracks the inband text tracks
+ *   @param {Array} metadataArray an array of meta data
+ * @private
+ */
+export const addDaterangeMetadata = ({
+  inbandTextTracks,
+  metadataArray,
+  timestampOffset
+}) => {
+  if (!metadataArray) {
+    return;
+  }
+
+  const Cue = window.WebKitDataCue || window.VTTCue;
+  const metadataTrack = inbandTextTracks.metadataTrack_;
+  const daterangeAttr = {
+    id: 'ID',
+    class: 'CLASS',
+    startDate: 'START-DATE',
+    duration: 'DURATION',
+    endDate: 'END-DATE',
+    endOnNext: 'END-ON-NEXT',
+    plannedDuration: 'PLANNED-DURATION',
+    scte35Out: 'SCTE35-OUT',
+    scte35In: 'SCTE35-IN'
+  };
+
+  if (!metadataTrack) {
+    return;
+  }
+
+  metadataArray.daterange.forEach((metadata) => {
+    const startTime = timestampOffset + (new Date(metadata.startDate).getTime() - new Date(metadataArray.dateTimeObject).getTime()) / 1000;
+    const endTime = metadata.endDate ? (new Date(metadata.endDate).getTime() - new Date(metadataArray.dateTimeObject).getTime()) / 1000 : 0;
+
+    const cue = new Cue(startTime, endTime, '');
+
+    cue.id = metadata.id;
+    Object.keys(metadata).forEach((key) => {
+      if (!['id', 'class', 'startDate', 'duration', 'endDate', 'endOnNext'].includes(key)) {
+        cue.value = {keys: daterangeAttr[key], data: metadata[key]};
```

Comment:
Should we map all hex values (eg: `SCTE-35`) to `ArrayBuffer` to mimic native Safari?